src/api/spec/factories/notification.rb

```diff
@@ -6,6 +6,19 @@
     title { Faker::Lorem.sentence }
     delivered { false }
 
+    transient do
+      originator { nil } # User login
+      recipient_group { nil } # Group title
+      role { nil } # Role title
+    end
+
+    after(:build) do |notification, evaluator|
+      notification.event_payload['who'] ||= evaluator.originator unless evaluator.originator.nil?
+      notification.event_payload['group'] ||= evaluator.recipient_group unless evaluator.recipient_group.nil?
+      notification.event_payload['role'] ||= evaluator.role unless evaluator.role.nil?
+      notification.event_payload['project'] ||= notification.notifiable.to_s unless notification.notifiable.nil?
```

Comment:
This line fills the `project` key no matter what type `notifiable` is.

Could we add a condition... or modify the introduced code to prevent assigning other types of notifiable to the `project` key?

Reply:
Good point, I'll adapt this.

Reply:
This is now only considering notifiables which are project and it won't crash even if the notifiable is `nil` since [`is_a?` is defined on Object](https://ruby-doc.org/core-3.1.2/Object.html#method-i-is_a-3F).